core/dbt/events/functions.py

```diff
@@ -344,3 +325,17 @@ def get_ts_rfc3339() -> str:
     ts = get_ts()
     ts_rfc3339 = ts.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
     return ts_rfc3339
+
+
+def add_to_event_history(event):
+    if flags.EVENT_BUFFER_SIZE == 0:
+        return
+    global EVENT_HISTORY
+    if EVENT_HISTORY is None:
+        reset_event_history()
+    EVENT_HISTORY.append(event)
```

Comment:
I stopped emitting the EventBufferFull event here because for small deque sizes we'd get strange behavior. What do we do if the deque is set to 1? to 2? Event at 10, we'd get a really high numberof EventBufferFull messages. I'm also not sure what the utility of the EventBufferFull message is.

Is this the right way to go?

Reply:
Agree that this message is oftentimes confusing / misleading for users. I do get why we added it, though. Let's keep it if EventBuffer is >=10000?

Reply:
Done.